### PR TITLE
cluster/buildx: cleanup background proxy with state management

### PR DESCRIPTION
**Pull request by gmichelo**
* State file to stable location
* Add node group name to metadata.json, so cleanup command can read it and avoid asking user for `--name` input
* Add background process ID to metadata.json, so cleanup command can read it and seng SIGTERM signal
* Implement SIGTERM handler to buildx proxy
* Print informative message even when buildx proxy is run in background
* Delete state folder when cleanup command is called
* Print error message if state metadata.json file is found

Fixes: NSL-1137

### CODE REVIEW

File: internal/cli/cmd/cluster/buildx.go [hunk 1]
```diff
@@ -56,8 +63,26 @@ func newSetupBuildxCmd(cmdName string) *cobra.Command {
 			return err
 		}
 
+		ctx, cancel := context.WithCancel(ctx)
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/buildx.go:66]: See also https://github.com/namespacelabs/foundation/blob/fe39ff7287121c5ef3da97c94055953046684c4b/internal/cli/fncobra/sigint.go#L16-L17 for an example of a composable version of this.

File: internal/cli/cmd/cluster/buildx.go [hunk 2]
```diff
@@ -67,34 +92,40 @@ func newSetupBuildxCmd(cmdName string) *cobra.Command {
...
+			NodeGroupName: *name,
+		}
 		for _, p := range available {
 			sockPath := filepath.Join(state, fmt.Sprintf("%s.sock", p))
 			md.Instances = append(md.Instances, buildxInstanceMetadata{
-				Platform:   p,
-				SocketPath: sockPath,
+				Platform:           p,
+				SocketPath:         sockPath,
+				BackgroundProxyPID: -1, // -1 means proxy is not in background
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/buildx.go:114]: pid 0 is also not defined so I'd lean on using that as the marker of pid not being set.

File: internal/cli/cmd/cluster/buildx.go [hunk 3]
```diff
@@ -157,26 +193,49 @@ func newSetupBuildxCmd(cmdName string) *cobra.Command {
...
+	NodeGroupName string                   `json:"node_group_name"`
+	Instances     []buildxInstanceMetadata `json:"instances"`
 }
 
 type buildxInstanceMetadata struct {
-	Platform   api.BuildPlatform `json:"build_platform"`
-	SocketPath string            `json:"socket_path"`
+	Platform           api.BuildPlatform `json:"build_platform"`
+	SocketPath         string            `json:"socket_path"`
+	BackgroundProxyPID int               `json:"bg_proxy_pid"`
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/buildx.go:238]: Nit: I think naming this simply "pid" would be fine.

File: internal/cli/cmd/cluster/buildx.go [hunk 4]
```diff
@@ -246,7 +305,40 @@ func newCleanupBuildxCommand() *cobra.Command {
...
+
+			for _, inst := range md.Instances {
+				if inst.BackgroundProxyPID > 0 {
+					process, err := os.FindProcess(inst.BackgroundProxyPID)
+					if err != nil {
+						return err
+					}
+
+					err = process.Signal(syscall.SIGTERM)
+					if err != nil && !errors.Is(err, os.ErrProcessDone) {
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/buildx.go:331]: Avoid re-assignments:

`if err := process.Signal(syscall.SIGTERM); err != nil && !errors.Is(err, ...) {`

File: internal/cli/cmd/cluster/buildx.go [hunk 5]
```diff
@@ -246,7 +305,40 @@ func newCleanupBuildxCommand() *cobra.Command {
...
+						return err
+					}
+				}
+			}
+
+			if err := os.RemoveAll(state); err != nil {
+				return err
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/buildx.go:338]: I'd make this a warning, to not skip txn.Remove as well.

File: internal/cli/cmd/cluster/buildx.go [hunk 6]
```diff
@@ -246,7 +305,40 @@ func newCleanupBuildxCommand() *cobra.Command {
...
+			}
+
+			return txn.Remove(md.NodeGroupName)
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/buildx.go:341]: `if md.NodeGroupName != "" {`?

File: internal/cli/cmd/cluster/buildx.go [hunk 7]
```diff
@@ -318,11 +410,19 @@ func banner(ctx context.Context, name string, use bool, native []api.BuildPlatfo
...
 
-	fmt.Fprintln(w)
-	fmt.Fprintln(w, style.Comment.Apply("Exiting will remove the configuration."))
+		fmt.Fprintln(w)
+		fmt.Fprintln(w, style.Comment.Apply("Exiting will remove the configuration."))
+	} else {
+		fmt.Fprintf(w, "\nStart building:\n")
+		fmt.Fprintf(w, "\n  docker buildx build ...\n")
+
+		fmt.Fprintf(w, "\nThe proxy is running in background. You can stop it and unconfigure the remote builders with:\n")
```
Comment (hugosantos) [inline internal/cli/cmd/cluster/buildx.go:423]: "proxy" is an implementation detail, I'd avoid making references to it. I suggest saying:

"Your remote builder context is running in the background. You can always clean up previous runs with: nsc docker buildx cleanup".

**Review by hugosantos — APPROVED** [review]

File: internal/cli/cmd/cluster/buildx.go (hunk 1 above)
Comment (gmichelo) [reply internal/cli/cmd/cluster/buildx.go:66]: Ah! Thanks for the reference. I see that `WithSigIntCancel` is already wired in 
https://github.com/namespacelabs/foundation/blob/cleanup-buildx-proxy/internal/cli/fncobra/do.go#L67
Which eventually is used by 
https://github.com/namespacelabs/foundation/blob/cleanup-buildx-proxy/internal/cli/fncobra/main.go#L62

So we can remove this bit and change the signal from SIGTERM to SIGINT.

**Pull request merged**